tensorflow_datasets/core/dataset_builder.py

```diff
@@ -205,6 +207,10 @@ def __init__(self, data_dir=None, config=None, version=None):
     else:  # Use the code version (do not restore data)
       self.info.initialize_from_bucket()
 
+    if self.OUTSIDE_TFDS:
+      path = os.path.dirname(inspect.getfile(self.__class__))
+      download.add_checksums_dir(path)
```

Comment:
Adding a checksums dir may have a significant overhead if many datasets are loaded at the same time. In addition, checksums are memoized, so new directories will be ignored.

Instead, we should update `download_and_prepare` or `DownloadManager` to check whether some `checksums.tsv` is present locally, and load checksums from this file if so. Otherwise, fallback to legacy checksums.